docs/src/main/asciidoc/security-authorization-of-web-endpoints-reference.adoc

```diff
@@ -265,82 +285,25 @@ public class SubjectExposingResource {
     }
 }
 ----
-<1> This `/subject/secured` endpoint requires an authenticated user that has been granted the role "Tester" through the use of the `@RolesAllowed("Tester")` annotation.
+<1> The `/subject/secured` endpoint requires an authenticated user with the granted "Tester" role through the use of the `@RolesAllowed("Tester")` annotation.
 <2> The endpoint obtains the user principal from the JAX-RS `SecurityContext`. This will be non-null for a secured endpoint.
 <3> The `/subject/unsecured` endpoint allows for unauthenticated access by specifying the `@PermitAll` annotation.
-<4> This call to obtain the user principal will return null if the caller is unauthenticated, non-null if the caller is authenticated.
-<5> The `/subject/denied` endpoint disallows any access regardless of whether the call is authenticated by specifying the `@DenyAll` annotation.
+<4> The call to obtain the user principal returns null if the caller is unauthenticated and non-null if the caller is authenticated.
+<5> The `/subject/denied` endpoint declares the `@DenyAll` annotation, thus disallowing all direct access to it as a REST method, regardless of the user calling it. The method is still invokable internally by other methods in this class.
 
-CAUTION: Please refer to the xref:security-built-in-authentication-support-concept.adoc#proactive-authentication[Proactive Authentication] section of the Built-In Authentication Support guide if you plan to use standard security annotations on IO thread.
+CAUTION: Please refer to the xref:security-built-in-authentication-support-concept.adoc#proactive-authentication[Proactive Authentication] section of the Built-In Authentication Support guide if you plan to use standard security annotations on the IO thread.
 
 The `@RolesAllowed` annotation value supports <<config-reference#property-expressions,Property Expressions>> including default values and nested Property Expressions.
 Configuration properties used with the annotation are resolved at runtime.
 
-[source,properties]
```

Comment:
Why this and another example after it are removed ?

Reply:
The purpose of the code example is purely to show how to use variables in the annotation value, I'd say a bullet list of examples (which is what Michal is adding below) is enough, no need for a code example that wraps the usage of variables into a lot of unrelated stuff.

Reply:
I highly agree

Reply:
@MichalMaler May be I'm missing something, can you paste an image showing how it will look like ?

Reply:
@sberyozkin I just rebased and applied all the fixes from yours and Jan review. Please have a look here, it should be rendered properly. If not, I will give you some build :)

https://github.com/quarkusio/quarkus/blob/0ef3d3fbf14702fb961d239b9d9c35c46338e528/docs/src/main/asciidoc/security-authorization-of-web-endpoints-reference.adoc

I think it is a huge simplification update.

Reply:
Not sure if I got the comment from Jan, but I don't agree with getting rid of the examples, unless the reworked doc will let the users click on the link and see the same example

Reply:
@sberyozkin If you look at https://github.com/quarkusio/quarkus/blob/2.16.0.Final/docs/src/main/asciidoc/security-authorization-of-web-endpoints-reference.adoc#authorization-using-annotations, you see that there are two snippets (`SubjectExposingResource` class) that are quite long and very similar. 

The purpose of the first one is to show how to use annotations. The purpose of the second snippet is specifically to show how to use Property expressions. I don't think that we need to copy (with minimal changes) the whole first snippet just to show how to use property expressions. Most of the code in that second snippet is irrelevant to property expressions.

Reply:
@jmartisk Nonetheless it is useful to have for users to copy and paste, a certain redundancy as far as the associated example code is not a problem IMHO

Reply:
Also seeing an example can help new users understand the idea better, IMHO the more examples the better

Reply:
Put the example back. My job is currently don. Now I need @sberyozkin to have a look on preview and give me a green light for the final peer review.
https://github.com/quarkusio/quarkus/blob/197069a60b2dc56265a36892b9e7165cb1d0bcbf/docs/src/main/asciidoc/security-authorization-of-web-endpoints-reference.adoc

Reply:
Thanks @MichalMaler